packages/calcite-components/src/tests/commonTests/utils.ts

```diff
@@ -74,3 +74,26 @@ export function getTagOrHTMLWithBeforeContent(componentTestSetup: TagOrHTML | Ta
 export function propToAttr(name: string): string {
   return name.replace(/([a-z0-9])([A-Z])/g, "$1-$2").toLowerCase();
 }
+
+export function validateCaretIndex({
+  page,
+  componentTag,
+  nestedInputTypeSelector = "input",
+  position,
+}: {
+  page: E2EPage;
+  componentTag: string;
+  nestedInputTypeSelector?: "textarea" | "input";
+  position?: number;
+}): Promise<boolean> {
+  return page.evaluate(
```

Comment:
Can we update the util to also do the assertion? We can rename it to `assertCaretPosition` to better reflect this.